### PR TITLE
[ExpandMemCmp] Optimize ExpandMemCmp to reduce instruction count on x86

### PR DESCRIPTION
**Pull request by igogo-x86**
Refactored the sequence of operations in MemCmpExpansion to zero-extend before byte-swapping. This change enables the generation of fewer instructions for x86, thereby improving code efficiency.

### CODE REVIEW

**Comment by topperc** [comment]
Wouldn't that byte swap the zeros that were added by the zext into the LSBs? I don't know this code, but that seems weird.

**Comment by igogo-x86** [comment]
> Wouldn't that byte swap the zeros that were added by the zext into the LSBs? I don't know this code, but that seems weird.

Yeah, they would, but since they are zeroes, they wouldn't affect the comparison result.

File: llvm/test/CodeGen/X86/memcmp.ll [hunk 1]
```diff
@@ -537,8 +529,8 @@ define i32 @length12(ptr %X, ptr %Y) nounwind {
 ; X64-NEXT:  # %bb.1: # %loadbb1
 ; X64-NEXT:    movl 8(%rdi), %ecx
 ; X64-NEXT:    movl 8(%rsi), %edx
-; X64-NEXT:    bswapl %ecx
-; X64-NEXT:    bswapl %edx
+; X64-NEXT:    bswapq %rcx
```
Comment (topperc) [inline llvm/test/CodeGen/X86/memcmp.ll:532]: bswapq is 2 uops vs 1 for bswapl on some CPU isn't it?

**Comment by igogo-x86** [comment]
Looks like it is not entirely profitable. I dropped it in favour of - https://github.com/llvm/llvm-project/pull/69942

**Pull request closed without merging**